packages/roosterjs-editor-api/lib/utils/applyInlineStyle.ts

```diff
@@ -50,18 +57,59 @@ export default function applyInlineStyle(
         editor.addUndoSnapshot(() => {
             let firstNode: Node;
             let lastNode: Node;
+            let rangeContainsOnlyTable: boolean = true;
+            const selectedCells = Array.from(getSelectedTableCells(editor));
+
             let contentTraverser = editor.getSelectionTraverser();
             let inlineElement = contentTraverser && contentTraverser.currentInlineElement;
+
             while (inlineElement) {
                 let nextInlineElement = contentTraverser.getNextInlineElement();
                 inlineElement.applyStyle((element, isInnerNode) => {
-                    callback(element, isInnerNode);
+                    const closestTD =

```

Comment:
too complex.

You just need to:

1. check if there is table selection. If so, get all selected cells (TD elements)
2. for each TD, editor.select(td);
3. Go with original code
4. if no table selection, go with original code.